### PR TITLE
fix: ModalPickerSingle should not overflow a fixed-width container

### LINKED ISSUE #365: ModalPickerSingle should not overflow a fixed-width container
When ModalPickerSingle is used within a fixed-width container like a table cell, a long label should not cause the button to be (partly) hidden.

### PR DESCRIPTION
**Pull request by gidomanders**
When ModalPickerSingle is used within a fixed-width container like a
table cell, a long label should not cause the button to be (partly)
hidden.

Closes #365

### CODE REVIEW

File: src/form/ModalPicker/ModalPickerOpener/useComponentOverflow/useComponentOverflow.test.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import * as useComponentOverflow from './useComponentOverflow';
+
+describe('HoC: useComponentOverflow', () => {
+  test('isOverflowing', () => {
+    expect(
+      useComponentOverflow.isOverflowing({
+        clientWidth: 50,
+        scrollWidth: 50
+      } as HTMLElement)
+    ).toBeFalsy();
```
Comment (MrHus) [inline src/form/ModalPicker/ModalPickerOpener/useComponentOverflow/useComponentOverflow.test.ts]: Don't use `toBeFalsy` or `toBeTruthy` just use toBe or toEqual and check the values directly

File: src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  const isOverflowing = useComponentOverflow(ref, values);
+
+  return (
+    <div className="d-flex align-items-center">
+      {values ? (
+        <span className="text-truncate mr-1 position-relative" ref={ref}>
+          {isOverflowing ? (
+            <Tooltip
+              content={values}
+              className="w-100 h-100 position-absolute truncator-tooltip"
```
Comment (MrHus) [inline src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.tsx]: What does the CSS class 'truncator-tooltip' do?

File: src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React, { useRef } from 'react';
```
Comment (MrHus) [inline src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.tsx]: Looks strange for the modal picker multiple: 

Screenshot 2020-03-18 at 13 09 54

File: src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.test.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import * as ModalPickerOpener from './ModalPickerOpener';
```
[the remark this answers is not in the document]
Comment (gidomanders) [reply src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.test.tsx]: This was a leftover from when I had to test a function inside the file, which was moved to a HoC later.

File: src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.tsx (hunk 2 above)
Comment (gidomanders) [reply src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.tsx]: Also a leftover from development, sorry for not checking for things like that

File: src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.tsx (hunk 3 above)
Comment (gidomanders) [reply src/form/ModalPicker/ModalPickerOpener/ModalPickerOpener.tsx]: Would you separate values and children to display children in the tooltip?

File: src/form/ModalPicker/single/ModalPickerSingle.test.tsx [hunk 5]
```diff
@@ -475,25 +480,35 @@ describe('Component: ModalPickerSingle', () => {
...
+      ).toBeUndefined();
 
       modalPickerSingle.setProps({
         value: adminUser
       });
 
-      const value = modalPickerSingle.find('span').text();
-      expect(value).toBe('admin@42.nl');
+      expect(
+        modalPickerSingle.find('ModalPickerOpener').props().values
```
Comment (MrHus) [inline src/form/ModalPicker/single/ModalPickerSingle.test.tsx]: Snapshots are a weapon for testing html, but it is a very crude tool. When possible use explicit value checks instead. Snapshots are a little too easy to approve.

**Pull request merged**